extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/JpaJandexScavenger.java

```diff
@@ -350,7 +350,8 @@ private void addClassHierarchyToReflectiveList(Collector collector, DotName clas
         // we need to check for enums
         for (FieldInfo fieldInfo : classInfo.fields()) {
             Type fieldType = fieldInfo.type();
-            if (Type.Kind.PRIMITIVE == fieldType.kind()) {
+            if (Type.Kind.CLASS != fieldType.kind()) {
```

Comment:
Hm, this would also skip parameterized types. I don't know if it could be a problem in the context of hibernate extension... just saying.

Reply:
This loop is about registering enum types for reflection. I don't think an enum can even be parameterized? Even if it can, I don't think the code below would work.

Regardless... I'm sure there are problems with this code, in particular with complex models involving generics. IMO our best way forward is to just avoid the need for registering types for reflection. Which is something we've been working on :)

Reply:
Ok np, I didn't know it's for enums only ;-)